consensus/core/src/tests/universal_committer_tests.rs

```diff
@@ -57,6 +58,272 @@ fn direct_commit() {
     };
 }
 
+/// Ensure idempotent replies.
+#[test]
+fn idempotence() {
+    telemetry_subscribers::init_for_testing();
+    // Commitee of 4 with even stake
+    let context = Arc::new(Context::new_for_test(4).0);
+    let dag_state = Arc::new(RwLock::new(DagState::new(
+        context.clone(),
+        Arc::new(MemStore::new()),
+    )));
+
+    // Create committer without pipelining and only 1 leader per round
+    let committer = UniversalCommitterBuilder::new(context.clone(), dag_state.clone()).build();
+
+    let decision_round_wave_2 = 5;
+    let references_decision_round_wave_2 = build_dag(
+        context.clone(),
+        dag_state.clone(),
+        None,
+        decision_round_wave_2,
+    );
+
+    // Commit one block.
+    let last_decided = Slot::new_for_test(0, 0);
+    let first_sequence = committer.try_commit(last_decided);
+    assert_eq!(first_sequence.len(), 1);
```

Comment:
Add a TODO to verify the round and authority index of committed leader? And which leader is committed here?

Reply:
Added the assertions for authority index & round. The leader committed should be the leader of wave 1. The variable names should help describe that.